### PR TITLE
feat: Create frontend support for AI Hint generation using ChatGPT for the basic layout

### PR DESCRIPTION
**Pull request by Mehul-Gandhi**
1. Create a renderGPTText helper function
2. Create frontend support for ChatGPT AI generated hints with streaming
3. Load the function URL through github secrets workflow
4. Load IAM AWS Access Key and Secret Access Key through github secret workflows
5. Support hint regeneration through a reload button

https://github.com/user-attachments/assets/14db8c8e-f50f-48db-ab35-fc163a896803

image

### CODE REVIEW

**Comment by Mehul-Gandhi** [comment]
Updated video with reloading hints

https://github.com/user-attachments/assets/79fcb04b-b8b0-4df5-b0b1-169346948588

File: .github/workflows/deploy-staging.yml [hunk 1]
```diff
@@ -58,6 +58,9 @@ jobs:
           REACT_APP_BUILD_TYPE: "platform-staging"
           REACT_APP_MIDDLEWARE_URL: ${{ secrets.STAGING_MIDDLEWARE_URL }}
           REACT_APP_COMMIT_HASH: ${{ github.sha }}
+          REACT_APP_AI_HINT_GENERATION_AWS_ENDPOINT: ${{ secrets.AI_HINT_GENERATION_AWS_ENDPOINT }}
+          REACT_APP_AWS_ACCESS_KEY: ${{ secrets.AWS_ACCESS_KEY }}
+          REACT_APP_AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_ACCESS_KEY }}
```
Comment (shreyabhandari0220) [inline .github/workflows/deploy-staging.yml:63]: Shouldn't this one say AWS_SECRET_ACCESS_KEY?

File: src/components/problem-layout/HintSystem.js [hunk 2]
```diff
@@ -38,7 +39,7 @@ class HintSystem extends React.Component {
         this.unlockFirstHint = props.unlockFirstHint;
         this.isIncorrect = props.isIncorrect;
         this.giveHintOnIncorrect = props.giveHintOnIncorrect
-
+        this.generateHintFromGPT = props.generateHintFromGPT;
```
Comment (shreyabhandari0220) [inline src/components/problem-layout/HintSystem.js:42]: Does the giveDynamicHint toggle determine the value this variable holds (i.e. is the value of props.generateHintFromGPT always the same as the giveDynamicHint toggle's value)?

File: src/components/problem-layout/ProblemCard.js [hunk 3]
```diff
@@ -382,96 +405,233 @@ class ProblemCard extends React.Component {
...
-        }
-
-        var quest = {
-            problem_title: this.problemTitle,
-            problem_subtitle: this.problemSubTitle,
-            question_title: this.step.stepTitle,
-            question_subtitle: this.step.stepBody,
-            student_answer: inputVal,
-            correct_answer: correctAnswer,
+        // var inputVal = "";
```
Comment (shreyabhandari0220) [inline src/components/problem-layout/ProblemCard.js:408]: You can remove the commented chunk of code

File: src/components/problem-layout/HintSystem.js (hunk 2 above)
Comment (Mehul-Gandhi) [reply src/components/problem-layout/HintSystem.js:42]: This is a function being passed in from `ProblemCard.js` to be called from `HintSystem.js` for the reload button

File: src/components/problem-layout/ProblemCard.js [hunk 4]
```diff
@@ -382,96 +405,233 @@ class ProblemCard extends React.Component {
...
+                    this.props.seed
+                ),
+                this.props.lesson,
+                this.props.courseName,
+                "dynamic",
+                this.state.dynamicHint,
+                this.state.bioInfo
+            );
+        };
+
```
Comment (shreyabhandari0220) [inline src/components/problem-layout/ProblemCard.js:634]: Quick question: are you logging 1) when the dynamic hint is opened and 2) when the regenerate button is clicked?

File: src/components/problem-layout/HintSystem.js (hunk 2 above)
Comment (shreyabhandari0220) [reply src/components/problem-layout/HintSystem.js:42]: Yeah disregard my comment, I reviewed `ProblemCard.js` after this.

File: .github/workflows/deploy-content-staging.yml [hunk 5]
```diff
@@ -106,7 +106,10 @@ jobs:
         REACT_APP_MIDDLEWARE_URL: ${{ secrets.STAGING_MIDDLEWARE_URL }}
         REACT_APP_COMMIT_HASH: ${{ github.sha }}
         REACT_APP_BUILD_TIMESTAMP: ${{ env.build_timestamp }}
-    
+        REACT_APP_AI_HINT_GENERATION_AWS_ENDPOINT: ${{ secrets.AI_HINT_GENERATION_AWS_ENDPOINT }}
+        REACT_APP_AWS_ACCESS_KEY: ${{ secrets.AWS_ACCESS_KEY }}
+        REACT_APP_AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_ACCESS_KEY }}
+
```
[the remark this answers is not in the document]
Comment (Mehul-Gandhi) [reply .github/workflows/deploy-content-staging.yml:112]: I fixed it

File: src/components/problem-layout/ProblemCard.js (hunk 4 above)
Comment (Mehul-Gandhi) [reply src/components/problem-layout/ProblemCard.js:610]: I added a TODO comment so we can do this in another PR

**Pull request merged**